lib/pat.c

```diff
@@ -1780,6 +1780,24 @@ delinfo_new(grn_ctx *ctx,
 
 /* pat operation */
 
+static inline uint32_t
+pat_node_get_w_of_pat_entry(uint32_t flags)
+{
+  size_t node_size;
+  if (flags & GRN_OBJ_KEY_LARGE) {
+    node_size = sizeof(pat_node_large);
+  } else {
+    node_size = sizeof(pat_node);
+  }
+
+  uint32_t w_of_pat_entry;
+  for (w_of_pat_entry = 0; (size_t)(1 << w_of_pat_entry) < node_size;
+       w_of_pat_entry++) {
+    /* nop */
+  }
```

Comment:
If we create a function to compute width of element, can https://github.com/groonga/groonga/blob/31a70fa45525b0fa533f572159c36b06084a4f02/lib/pat.c#L1794-L1797 also reuse the extracted bit width computation feature?